adding store mapping parameter docs (#9578)
opensearch-project/documentation-website: _field-types/mapping-parameters/store.md

---
layout: default
title: Store
parent: Mapping parameters
grand_parent: Mapping and field types
nav_order: 180
has_children: false
has_toc: false
---

# Store

The `store` mapping parameter determines whether the value of a field should be stored separately from the `_source` and made directly retrievable using the `stored_fields` option in a search request.

> By default, `store` is set to `false`, meaning field values are not stored individually and are only available as part of the `_source` document. If `store` is set to `true`, you can disable the `_source` to save disk space and still retrieve specific fields.

Reviewer: ```suggestion
By default, `store` is set to `false`, meaning field values are not stored individually and are only available as part of the document `_source`. If `store` is set to `true`, you can disable the `_source` to save disk space and still retrieve specific fields.
```

> 

Reviewer: In line 15, could you link to retrieve specific fields? https://opensearch.org/docs/latest/search-plugins/searching-data/retrieve-specific-fields/

Author: Thats done now

> By default, `store` is set to `false`, meaning field values are not stored individually and are only available as part of the document `_source`. If `store` is set to `true`, you can disable the `_source` to save disk space and still [retrieve specific fields]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/retrieve-specific-fields/).

Reviewer 2: ```suggestion
By default, `store` is set to `false`, meaning that field values are not stored individually and are only available as part of the document `_source`. If `store` is set to `true`, you can disable the `_source` to save disk space and still [retrieve specific fields]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/retrieve-specific-fields/).
```

> ## Example: Enabling store on a field

Reviewer 2: ```suggestion
## Example: Enabling `store` on a field
```

> The following request creates an index named `products` where the `model` field is stored separately from the `_source`:

Reviewer: ```suggestion
The following request creates an index named `products`, in which the `model` field is stored separately from the `_source`:
```

> The following request creates an index named `products`, in which the `model` field is stored separately from the `_source`:

Reviewer 2: ```suggestion
The following request creates an index named `products` in which the `model` field is stored separately from the `_source`:
```

```json
PUT /products
{
  "mappings": {
    "properties": {
      "model": {
        "type": "keyword",
        "store": true
      },
      "name": {
        "type": "text"
      }
    }
  }
}
```
{% include copy-curl.html %}

> ### Indexing a document

Reviewer: ```suggestion
Ingest a document into the index:
```

```json
PUT /products/_doc/1
{
  "model": "WM-1001",
  "name": "Wireless Mouse"
}
```
{% include copy-curl.html %}

> ### Retrieving only the stored field

Reviewer: ```suggestion
Retrieve only the stored field:
```

```json
POST /products/_search
{
  "query": {
    "match": {
      "name": "Mouse"
    }
  },
  "stored_fields": ["model"]
}
```
{% include copy-curl.html %}

> This returns the `model` field stored separately, even though `_source` is still available.

Reviewer: ```suggestion
This query returns the `model` field stored separately, even though `_source` is still available.
```

> This query returns the `model` field stored separately, even though `_source` is still available.

Reviewer 2: ```suggestion
This query returns the `model` field stored separately even though the `_source` is still available.
```

---

## Example: Storing fields with `_source` disabled

> If you want to save disk space and don't need the full original document later (e.g., for reindexing or updates), you can disable `_source` and store only necessary fields:

Reviewer: ```suggestion
If you want to save disk space and don't need the full original document later (for example, for reindexing or updates), you can disable `_source` and store only necessary fields:
```

> If you want to save disk space and don't need the full original document later (for example, for reindexing or updates), you can disable `_source` and store only necessary fields:

Reviewer 2: ```suggestion
If you want to save disk space and don't need to access the full original document later (for example, for reindexing or updates), you can disable `_source` and store only necessary fields:
```

```json
PUT /products_no_source
{
  "mappings": {
    "_source": {
      "enabled": false
    },
    "properties": {
      "model": {
        "type": "keyword",
        "store": true
      },
      "name": {
        "type": "text"
      }
    }
  }
}
```
{% include copy-curl.html %}

> ### Indexing a document

Reviewer: ```suggestion
Ingest a document into the index:
```

```json
PUT /products_no_source/_doc/1
{
  "model": "KB-2002",
  "name": "Mechanical Keyboard"
}
```
{% include copy-curl.html %}

> ### Retrieving the stored field

Reviewer: ```suggestion
Retrieve the stored field:
```

```json
POST /products_no_source/_search
{
  "query": {
    "match": {
      "name": "Keyboard"
    }
  },
  "stored_fields": ["model"]
}
```
{% include copy-curl.html %}

> This shows the `model` field retrieved from stored fields without accessing `_source`.

Reviewer: ```suggestion
This query returns the `model` field retrieved from stored fields without accessing `_source`.
```

> ### Attempting to retrieve the `_source`

Reviewer: ```suggestion
If you attempt to retrieve the `_source` as follows:
```

> This query returns the `model` field retrieved from stored fields without accessing `_source`.

Reviewer 2: ```suggestion
This query returns the `model` field retrieved from stored fields without accessing the `_source`.
```

> 

Reviewer 2: Line 120: Should "stored fields" be either "the stored fields" or `stored_fields`?

Author: both are valid here I think, but I'll change to stored_fields

```json
GET /products_no_source/_doc/1
```

> Since `_source` is disabled, the response demonstrates that the full document is no longer available and only stored fields can be retrieved:

Reviewer: ```suggestion
The `_source` in the response will be `null`. This demonstrates that the full document is no longer available and only stored fields can be retrieved because `_source` is disabled:
```

> The `_source` in the response will be `null`. This demonstrates that the full document is no longer available and only stored fields can be retrieved because `_source` is disabled:

Reviewer 2: ```suggestion
Then the `_source` in the response will be `null`. This demonstrates that the full document is no longer available and that only stored fields can be retrieved because `_source` is disabled:
```

---
layout: default
title: Store
parent: Mapping parameters
grand_parent: Mapping and field types
nav_order: 180
has_children: false
has_toc: false
---

# Store

The `store` mapping parameter determines whether the value of a field should be stored separately from the `_source` and made directly retrievable using the `stored_fields` option in a search request.

> By default, `store` is set to `false`, therefore field values are not stored individually, they are only available as part of the `_source` document. If `store` is set to `true`, you can exclude the `_source` to save disk space but still fetch certain fields.

Reviewer 3: Can we give an example for source disabled as well ?
